File: frontend/app/models/pbx-oncall-shift.js
import DS from 'ember-data'
import pbxOncallShiftValidations from '../validations/pbxOncallShift'


export default DS.Model.extend(pbxOncallShiftValidations, {
  dayFri: DS.attr('boolean', {defaultValue: false}),
  dayMon: DS.attr('boolean', {defaultValue: false}),
  daySat: DS.attr('boolean', {defaultValue: false}),
  daySun: DS.attr('boolean', {defaultValue: false}),
  dayThu: DS.attr('boolean', {defaultValue: false}),
  dayTue: DS.attr('boolean', {defaultValue: false}),
  dayWed: DS.attr('boolean', {defaultValue: false}),
  metric: DS.attr(),
  name: DS.attr(),
  pbxOncallId: DS.attr(), //Display Only
  priority: DS.attr('number'),
  startTime: DS.attr(),
  endTime: DS.attr(),
  startAt: DS.attr(),
  endAt: DS.attr(),
  pbxOnCall: DS.belongsTo('pbxOnCall'),
  pbxOncallShiftUsers: DS.hasMany('pbxOncallShiftUser'),
})

File: app/models/pbx_on_call_shift.rb
class PbxOnCallShift < RestModel
    include ActiveModel::Validations

    embeds_many :pbx_oncall_shift_users, class_name: :pbx_on_call_shift_user

    property :day_fri
    property :day_mon
    property :day_sat
    property :day_sun
    property :day_thu
    property :day_tue
    property :day_wed
    property :end_at
    property :end_time
    property :id
    property :metric
    property :name
    property :pbx_oncall_id #Display Only
    property :priority
    property :start_time
    property :start_at

    validates :name, presence: true

    def self.load_from_json(json)
        PbxOnCallShift.from_source(JSON.parse(json)).first
    end

end